assets/js/components/notifications/EnableAutoUpdateBannerNotification.test.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * InternalServerError component tests.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	render,
+	createTestRegistry,
+	screen,
+	waitFor,
+	act,
+	fireEvent,
+} from '../../../../tests/js/test-utils';
+import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import EnableAutoUpdateBannerNotification from './EnableAutoUpdateBannerNotification';
+import API from 'googlesitekit-api';
+
+import mockUseQueryArg from '../../hooks/useQueryArg';
+jest.mock( '../../hooks/useQueryArg' );
+
+// setup mockImplementation for `useQueryArg`
+function stubMockUseQueryArg( isNewPluginInstall = false ) {
+	mockUseQueryArg.mockImplementation( ( queryArg ) => {
+		if ( isNewPluginInstall && queryArg === 'notification' ) {
+			return [ 'authentication_success' ];
+		}
+		return [ false ];
+	} );
+}
+
+describe( 'EnableAutoUpdateBannerNotification', () => {
+	const registry = createTestRegistry();
+
+	describe( 'existing site kit setup', () => {
+		beforeEach( async () => {
+			jest.resetAllMocks();
+			stubMockUseQueryArg();
+
+			jest.spyOn( API, 'getItem' ).mockImplementation( () => {
+				return Promise.resolve( { cacheHit: false } );
+			} );
+
+			jest.spyOn( API, 'setItem' ).mockImplementation( () => {
+				return Promise.resolve( { cacheHit: false } );
+			} );
```

Comment:
That's not the return value/format of `setItem`; if we're mocking functions we should make sure the mocks are the same return values as the real function would be. `setItem` always returns a Promise that resolves with a boolean: https://github.com/google/site-kit-wp/blob/3b347f5566e5e725392c68e99348d01e7ad639d6/assets/js/googlesitekit/api/cache.js#L215-L279

Also, `mockImplementation` allows for async mocks, so we don't need to write `Promise.resolve` here:

```suggestion
			jest.spyOn( API, 'setItem' ).mockImplementation( async () => {
				return true;
			} );
```

Reply:
The suggested change is against the eslint rules, so I changed it to:
```js
jest.spyOn( API, 'setItem' ).mockImplementation( () => {
	return Promise.resolve( true );
} );
```